src/utils/storage.ts

```diff
@@ -410,14 +402,9 @@ export async function processHistory(
  * @returns {Promise<void>}
  */
 export async function _updateTokensData(storage: IStorage, tokens: Set<string>): Promise<void> {
-  async function fetchTokenData(uid: string): Promise<
-    | {
-        success: true;
-        name: string;
-        symbol: string;
-      }
-    | { success: false; message: string }
-  > {
+  async function fetchTokenData(
+    uid: string
+  ): Promise<GeneralTokenInfoSchema | { success: true; name: string; symbol: string }> {
```

Comment:
Yes, because below there's a return with this explicit type, which does not conform to `GeneralTokenInfoSchema`:

https://github.com/HathorNetwork/hathor-wallet-lib/blob/ed121c6a6a0a8f0db7abda2626b640a60a8f58df/src/utils/storage.ts#L412-L416

This function could definitely be improved, but I think this is a more general typing issue that should be addressed in separate PRs.